scripts/verify-pkg-main.js

```diff
@@ -36,31 +36,73 @@ if (!isValidCwd) {
 }
 
 let invalidMains = 0;
+let invalidModules = 0;
+let invalidTypes = 0;
 globSync('packages/*/package.json').forEach((jsonPath) => {
-  const packageInfo = JSON.parse(fs.readFileSync(jsonPath));
-  if (!packageInfo.main) {
+  const packageJson = JSON.parse(fs.readFileSync(jsonPath));
+  if (!packageJson.main) {
     return;
   }
+  verifyPath(packageJson, jsonPath, 'main');
+  verifyPath(packageJson, jsonPath, 'module');
+  verifyPath(packageJson, jsonPath, 'types');
+});
+
+if (invalidMains > 0 || invalidModules > 0 || invalidTypes > 0) {
+  if (invalidMains > 0) {
+    logErrorAndExit(`${invalidMains} incorrect main property values found; please fix.`);
+  }
+  if (invalidModules > 0) {
+    logErrorAndExit(`${invalidModules} incorrect module property values found; please fix.`);
+  }
+  if (invalidTypes > 0) {
+    logErrorAndExit(`${invalidTypes} incorrect types property values found; please fix.`);
+  }
+} else {
```

Comment:
You can remove the if/else now (as illustrated in Andy's original comment) since this line will only be reached if none of the above `if`s were true.

Reply:
Actually... Andy has a more valid point here, that we shouldn't be using logErrorAndExit at all, so ignore my comment here